server.js

```diff
@@ -1,38 +1,258 @@
-// import topMusicData from './data/top-music.json'
-
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
-mongoose.Promise = Promise
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
+import express from 'express';
+import bodyParser from 'body-parser';
+import cors from 'cors';
+import mongoose from 'mongoose';
+import booksData from './data/books.json';
+import authorsData from './data/authors.json';
+
+const mongoUrl = process.env.MONGO_URL || 'mongodb://localhost/project-mongo';
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
+mongoose.Promise = Promise;
+
+// Create my database model's
+const Book = mongoose.model('Book', {
+  bookID: Number,
+  title: String,
+  authors: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'Author'
+  },
+  average_rating: Number,
+  isbn: String,
+  isbn13: String,
+  language_code: String,
+  num_pages: Number,
+  ratings_count: Number,
+  text_reviews_count: Number
+});
+
+const Author = mongoose.model('Author', {
+  authors: String
+});
+
+if (process.env.RESET_DATABASE) {
+  // POPULATING DATABASE WITH TWO COLLECTIONS (WITH RELATIONS)
+  const populateDatabase = async () => {
+    // First of all, clear current content of two collections
+    await Book.deleteMany();
+    await Author.deleteMany();
+
+    // Next, declare empty array in which later on will
+    // store all authors from authors.json (of Author models)
+    let authorsArray = [];
+
+    authorsData.forEach(async (item) => {
+      const newAuthor = new Author(item);
+
+      // Push each newAuthor to array authorsArray
+      authorsArray.push(newAuthor);
+      await newAuthor.save();
+    });
+
+    // Create new book for element in booksData array from JSON file.
+    // Important thing to notice: in JSON file we had property "authors" with
+    // hardcoded string value. We need it to detect which author model should
+    // each book have. Later on, hardcoded "authors" property will be
+    // overwritten by new "authors" property, the one with value of ObjectId type.
+    // For further reference on that, check out last example from website below,
+    // the one about keys collision : https://davidwalsh.name/merge-objects
+    booksData.forEach(async (bookItem) => {
+      const newBook = new Book({
+        ...bookItem,
+        authors: authorsArray.find(
+          (authorItem) => authorItem.authors === bookItem.authors
+        )
+      });
+      await newBook.save();
+    });
+  };
+  populateDatabase();
+}
+
+// Defines the port the app will run on. Defaults to 8080, but can be
 // overridden when starting the server. For example:
 //
 //   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+const port = process.env.PORT || 8080;
+const app = express();
+
+const myEndpoints = require('express-list-endpoints');
```

Comment:
You can also `import endpoints from 'express-list-endpoints` and use it as a variable, if you'd like to! 😊